File: 21-merge-two-sorted-lists/merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def mergeTwoLists(self, list1, list2):
        """
        :type list1: Optional[ListNode]
        :type list2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        temp1=list1
        temp2=list2
        lst1=[]
        lst2=[]
        while temp1:
            lst1.append(temp1.val)
            temp1=temp1.next

        while temp2:
            lst2.append(temp2.val)
            temp2=temp2.next

        lst1=lst1+lst2
        lst1.sort()

        dummy=ListNode(0)
        t=dummy
        for x in lst1:
            t.next=ListNode(x)
            t=t.next

        return dummy.next

File: 21-merge-two-sorted-lists/merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode temp1=list1;
        ListNode temp2=list2;
        List<Integer> lst1=new ArrayList<>();
        List<Integer> lst2=new ArrayList<>();
        // List<Integer> res=new ArrayList<>();
        while(temp1!=null){
            lst1.add(temp1.val);
            temp1=temp1.next;
        }
         while(temp2!=null){
            lst2.add(temp2.val);
            temp2=temp2.next;
        }
        lst1.addAll(lst2);
        // for(int i=0;i<Math.max(list1.size(),list2.size());i++){
        //     if(i<list1.size()) res.add(list1.get(i));//1 2 4
        //     if(i<list2.size()) res.add(list2.get(i));//1 3 4
        // }
        Collections.sort(lst1);
        ListNode dummy=new ListNode(0);
        ListNode t=dummy;
        for(int l:lst1){
            t.next=new ListNode(l);
            t=t.next;
        }
        return dummy.next;        
    }
}